src/decorators/index.ts

```diff
@@ -44,36 +48,37 @@ const verifyJwt = (tokenJwtString: string): TokenValidationResult => {
   if (key.trim() === '') {
     const err = 'no key found to verify the token';
     L.debug(err);
-    return { success: false, data: err };
+    return { success: false };
   }
   try {
-    const decoded: TokenData = jwt.verify(`${tokenJwtString}`, key);
-    return { success: true, data: decoded };
+    const decoded = jwt.verify(`${tokenJwtString}`, key);
+    const scopes = _.isObjectLike(decoded) && (<JwtPayload>decoded).context?.scopes;
+    return _.isArray(scopes) ? { success: true, data: { scopes } } : { success: false };
   } catch (err) {
     L.debug(`invalid token provided ${err}`);
-    return { success: false, data: err };
+    return { success: false };
   }
 };
 
-const verifyEgoApiKey = async (uuidString: string): Promise<TokenValidationResult> => {
+const verifyEgoApiKey = async (keyString: string): Promise<TokenValidationResult> => {
   const EGO_URL = config.getConfig().egoUrl();
   const EGO_CLIENT_ID = config.getConfig().egoClientId();
   const EGO_CLIENT_SECRET = config.getConfig().egoClientSecret();
   const auth = 'Basic ' + Buffer.from(EGO_CLIENT_ID + ':' + EGO_CLIENT_SECRET).toString('base64');
 
-  return await fetch(`${EGO_URL}/o/check_api_key?apiKey=${uuidString}`, {
+  return await fetch(`${EGO_URL}/o/check_api_key?apiKey=${keyString}`, {
     method: 'post',
     headers: { 'Content-Type': 'application/json', authorization: auth },
   })
     .then(async res => {
       const token = await res.json();
-
+      const { scopes } = token;
       if (token.error) throw token.error;
-      return { success: true, data: token };
+      return { success: true, data: { scopes } };
```

Comment:
what if scopes is undefined?

Reply:
I'll update conditional `if (token.error || !Boolean(scopes))`

Reply:
Updated

Reply:
`!Boolean(scopes)` is fine, I usually see this written like `!!scopes`

Reply:
`!scopes` might be sufficient. why do you use the `Boolean()` function?

Reply:
Eh just been using Boolean + other primitives a little more frequently after some recent code review suggestions, in this case it really is just stylistic. Updated conditional to `!scopes`